pkg/controller/labelidentity/label_group_index.go

```diff
@@ -0,0 +1,510 @@
+		for _, labelIdentityKeys := range i.labelIdentityNamespaceIndex {
+			i.scanLabelIdentityMatches(labelIdentityKeys, sItem)
+		}
+	}
+	return i.getMatchedLabelIdentityIDs(sItem)
+}
+
+// DeleteSelector removes a selectorItem from referring to the policy being deleted.
+func (i *LabelIdentityIndex) DeleteSelector(selectorKey string, policyKey string) {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	i.deleteSelector(selectorKey, policyKey)
+}
+
+func (i *LabelIdentityIndex) deleteSelector(selectorKey string, policyKey string) {
+	s, exists, _ := i.selectorItems.GetByKey(selectorKey)
+	if !exists {
+		return
+	}
+	sItem := s.(*selectorItem)
+	if sItem.policyKeys.Equal(sets.NewString(policyKey)) {
+		// delete the selectorItem and any LabelIdentity mappings if there's no
+		// policy left that has the selector anymore.
+		for lkey := range sItem.labelIdentityKeys {
+			labelIdentity := i.labelIdentities[lkey]
+			labelIdentity.selectorItemKeys.Delete(selectorKey)
+		}
+		i.selectorItems.Delete(sItem)
+	} else {
+		i.updateSelectorItem(sItem, selectorMatchedPolicyDelete, policyKey)
+	}
+}
+
+// SetPolicySelectors registers ClusterSet-scope policy selectors with the labelIdentityIndex,
+// and then retrieves all the LabelIdentity IDs that currently match these selectors.
+func (i *LabelIdentityIndex) SetPolicySelectors(selectors []*types.GroupSelector, policyKey string) []uint32 {
+	var labelIdentityIDs []uint32
+	newSelectors := map[string]*types.GroupSelector{}
+	for _, s := range selectors {
+		klog.V(2).InfoS("Getting matched LabelIdentity for policy selector", "selector", s.NormalizedName, "policy", policyKey)
+		newSelectors[s.NormalizedName] = s
+	}
+	originalSelectors := i.getPolicySelectors(policyKey)
+	if len(originalSelectors) > 0 {
+		var matchedSelectorKeys []string
+		for selKey := range originalSelectors {
+			if _, exists := newSelectors[selKey]; exists {
+				// These clusterset-scoped selectors are already bound to the policy in labelIdentityIndex.
+				// We can simply read matched label identity IDs from the index.
+				selectedLabelIDs := i.GetLabelIdentityIDs(selKey)
+				labelIdentityIDs = append(labelIdentityIDs, selectedLabelIDs...)
+				matchedSelectorKeys = append(matchedSelectorKeys, selKey)
+			}
+		}
+		// Remove matched clusterset-scoped selectors of the policy before and after the update.
+		// The selectors remaining in newSelectors will be new ones to be registered for the policy
+		// The selectors remaining in originalSelectors will need to be removed from the labelIdentityIndex
+		for _, k := range matchedSelectorKeys {
+			delete(originalSelectors, k)
+			delete(newSelectors, k)
+		}
+	}
+	for _, sel := range newSelectors {
+		selectedLabelIDs := i.AddSelector(sel, policyKey)
+		labelIdentityIDs = append(labelIdentityIDs, selectedLabelIDs...)
+	}
+	for selectorKey := range originalSelectors {
+		i.DeleteSelector(selectorKey, policyKey)
+	}
+	// Dedup labelIdentityIDs in-place
```

Comment:
Keep the description the same as other lines:
```suggestion
	// Dedup label Identity IDs in-place.
```